projects/packages/publicize/tests/php/test-social-image-generator/test-extractor.php

```diff
@@ -0,0 +1,167 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+/**
+ * Testing the Extractor.
+ *
+ * @package automattic/jetpack-publicize
+ */
+
+namespace Automattic\Jetpack\Publicize;
+
+use WorDBless\BaseTestCase;
+
+/**
+ * Testing the SIG Extractor.
+ */
+class Extractor_Test extends BaseTestCase {
+	/**
+	 * Post ID of the testing post.
+	 *
+	 * @var int $post_id
+	 */
+	protected $post_id;
+
+	/**
+	 * ID of the test attachment.
+	 *
+	 * @var int $attachment_id
+	 */
+	protected $attachment_id;
+
+	/**
+	 * Instance of the Extractor class used for testing.
+	 *
+	 * @var Extractor Instance of the Extractor class.
+	 */
+	protected $extractor;
+
+	/**
+	 * Initialize tests
+	 *
+	 * @before
+	 */
+	public function set_up() {
+		$this->post_id       = wp_insert_post(
+			array(
+				'post_title'   => 'hello',
+				'post_content' => 'world',
+				'post_status'  => 'publish',
+			)
+		);
+		$this->extractor     = new Social_Image_Generator\Extractor( $this->post_id );
+		$this->attachment_id = $this->create_upload_object( realpath( __DIR__ . '/..' ) . '/images/jetpack-logo.png' );
+	}
+
+	/**
+	 * Update the settings for the image generator.
+	 *
+	 * @param array $value Array of settings to update.
+	 */
+	private function update_image_generator_settings( $value ) {
+		update_post_meta( $this->post_id, Publicize::POST_JETPACK_SOCIAL_OPTIONS, array( 'image_generator_settings' => $value ) );
+	}
+
+	/**
+	 * Reset settings after each test.
+	 */
+	public function tear_down() {
+		$this->update_image_generator_settings( array() );
+	}
+
+	/**
```

Comment:
This method is copied from a test in Social: https://github.com/Automattic/jetpack/blob/b313bd253afdf919adf7a71a866d0ab85b3a7bda/projects/plugins/social/tests/php/test-class-meta-tags.php#L63.

It would be good to add some standard utilities that can be used across plugins and packages.